base/threadcall.jl

```diff
@@ -0,0 +1,84 @@
+const max_ccall_threads = parse(Int, get(ENV, "UV_THREADPOOL_SIZE", "4"))
+const thread_notifiers = [Nullable{Condition}() for i in 1:max_ccall_threads]
```

Comment:
It does appear that libuv has a single global thread pool. I would suggest documenting the same and we only use UV_THREADPOOL_SIZE - 2 threads? `threadcall_restrictor` limits the number of concurrent `@threadcall` calls active.